code/game/machinery/poolcontroller.dm

```diff
@@ -136,4 +89,43 @@
 /obj/machinery/poolcontroller/proc/mistoff() //Delete all /obj/effect/mist from all linked pool tiles.
 	for(var/obj/effect/mist/M in linkedmist)
 		del(M)
-	misted = 0 //no mist left, turn off the tracking var
\ No newline at end of file
+	misted = 0 //no mist left, turn off the tracking var
+
+/obj/machinery/poolcontroller/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	var/data[0]
+
+	data["currentTemp"] = capitalize(src.temperature)
+	data["emagged"] = emagged
+	data["TempColor"] = temperaturecolor
+
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "poolcontroller.tmpl", "Pool Controller Interface", 520, 410)
+		ui.set_initial_data(data)
+		ui.open()
+
+/obj/machinery/poolcontroller/Topic(href, href_list)
```

Comment:
will point out this is fixed on the next line, but github being github means this still appears